packages/extension/src/inpage/inpage.model.ts

```diff
@@ -58,18 +74,26 @@ interface IStarketWindowObject {
   ) => Promise<T["result"]>
   enable: (options?: { showModal?: boolean }) => Promise<string[]>
   isPreauthorized: () => Promise<boolean>
-  on: (event: "accountsChanged", handleEvent: EventHandler) => void
-  off: (event: "accountsChanged", handleEvent: EventHandler) => void
+  on: <T extends WalletEvents>(
+    event: T["type"],
+    handleEvent: T["handler"],
+  ) => void
+  off: <T extends WalletEvents>(
+    event: T["type"],
+    handleEvent: T["handler"],
+  ) => void
```

Comment:
better type inference where? why not do it like this?
```suggestion
  on: (
    event: WalletEvents["type"],
    handleEvent: WalletEvents["handler"],
  ) => void
  off: (
    event: WalletEvents["type"],
    handleEvent: WalletEvents["handler"],
  ) => void
```